sdk/client/resourcemanager/poller_lro.go

```diff
@@ -170,11 +169,41 @@ func (p *longRunningOperationPoller) Poll(ctx context.Context) (result *pollers.
 		for k, v := range statuses {
 			if strings.EqualFold(string(op.Properties.ProvisioningState), string(k)) {
 				result.Status = v
-				return
+				break
 			}
 			if strings.EqualFold(string(op.Status), string(k)) {
 				result.Status = v
-				return
+				break
+			}
+		}
+
+		if result.Status == pollers.PollingStatusFailed {
+			apiResponse := http.Response{
+				Body: io.NopCloser(bytes.NewReader(respBody)),
+			}
+			lroError, parseError := parseErrorFromApiResponse(apiResponse)
```

Comment:
we should be able to pass in `result.HttpResponse` here instead, although we'd need to re-assign the body above to account for this (which we should be doing anyway, but since we haven't needed to reuse the http body, it's not present until now) - meaning that line ~118 needs:

```
result.HttpResponse.Body = io.NopCloser(bytes.NewReader(respBody))
```

and then we should be able to use `result.HttpResponse` directly here?